video/out/present_sync.c

```diff
@@ -71,8 +71,8 @@ void present_sync_swap(struct mp_present *present)
         return;
     }
 
-    uint64_t now_monotonic = ts.tv_sec * 1000000LL + ts.tv_nsec / 1000;
-    uint64_t ust_mp_time = mp_time_us() - (now_monotonic - ust);
+    uint64_t now_monotonic = ts.tv_sec * UINT64_C(1000000000) + ts.tv_nsec;
+    uint64_t ust_mp_time = mp_time_ns() - (now_monotonic - ust);
```

Comment:
this changes the assumption on what unit `ust` is from microseconds to nanoseconds. intended?

Reply:
This is fully part of extending vsync precision. `present_sync` is fully controlled by us and is feed from wayland/x11 presentation feedback. `ust` here is defined by mpv and it is this commit goal to change it to ns. Can't really split it up without breaking things.

Reply:
I was wrong earlier, the 1000 multiplication you did for xorg was correct. I did some reading and wayland's presentation time nanosecond part is extra. The UST part (tv_sec_lo + tv_sec_hi) is what's equivalent to xorg and I guess it is microseconds. Or well that's what we've been assuming at least it works.

Reply:
Actually, I realized that converting `ust` to ns breaks a lot of other assumptions. e.g. wayland uses the calculated vsync_duration value and the `ust_passed/msc_passed` part only works if ust is microseconds.

Reply:
Yes, I was testing on Wayland briefly today and it like you say doesn't work. This patch is half baked. Will need to properly test everything. `vsync_duration` is fine actually, the other one is messed up. Don't worry about it for now, will do proper patch.

Reply:
You could just leave the present sync stuff in microseconds. That's basically what you did for drm and it works there. Honestly, there's no reason drm can't also use the present_sync codepath. I'm pretty sure the sbc stuff useless, but that's something for another day.

Reply:
on x11 `ust` comes directly from `XPresentCompleteNotifyEvent` and you did not modify that part of code.
so either:
1. the new assumption is that x11 provides microseconds
2. this change here is incorrect